lib/routes/home/wallet_dashboard.dart

```diff
@@ -114,7 +102,38 @@ class WalletDashboardState extends State<WalletDashboard> {
                                               widget._offsetFactor)))
                   : SizedBox(),
             ),
-          )
+          ),
+          Positioned(
+              top: 70 - BALANCE_OFFSET_TRANSITION * widget._offsetFactor,
+              child: Center(
+                child: widget._accountModel != null &&
+                        !widget._accountModel.initial &&
+                        (widget._accountModel.fiatConversionList.isNotEmpty ||
+                            widget._accountModel.fiatCurrency != null) &&
+                        isAboveMinAmount(widget._accountModel.fiatCurrency)
+                    ? FlatButton(
+                        onPressed: () {
+                          var nextFiatCurrencyIndex = widget
+                                  ._accountModel.fiatConversionList
+                                  .indexOf(widget._accountModel.fiatCurrency) +
+                              1;
+                          findNextValidFiatConversion(nextFiatCurrencyIndex %
```

Comment:
The name of `findNextValidFiatConversion` implies read-only. Better to keep it read-only and call the `_onFiatCurrencyChange` callback here.